File: Ruby/example_usage.rb
# example_usage.rb
# Example usage of the calculate_eigenvalues and calculate_eigenvector functions

require_relative 'eigen_calculation'

# Define a 2x2 matrix
matrix = Matrix[[4, 1], [2, 3]]

# Calculate eigenvalues
eigenvalues = calculate_eigenvalues(matrix)
puts "Eigenvalues:"
eigenvalues.each { |val| puts val }

# Calculate eigenvectors for each eigenvalue
puts "\nEigenvectors:"
eigenvalues.each do |eigenvalue|
  eigenvector = calculate_eigenvector(matrix, eigenvalue)
  puts "Eigenvector for eigenvalue #{eigenvalue}: #{eigenvector}"
end

File: PHP/example_usage.php
<?php
// example_usage.php
// Example usage of the calculateEigenvalues and calculateEigenvector functions

require_once 'eigen_calculation.php';

// Define a 2x2 matrix
$matrix = [
    [4, 1],
    [2, 3]
];

// Calculate eigenvalues
$eigenvalues = calculateEigenvalues($matrix);
echo "Eigenvalues:\n";
foreach ($eigenvalues as $val) {
    echo $val . "\n";
}

// Calculate eigenvectors for each eigenvalue
echo "\nEigenvectors:\n";
foreach ($eigenvalues as $eigenvalue) {
    $eigenvector = calculateEigenvector($matrix, $eigenvalue);
    echo "Eigenvector for eigenvalue $eigenvalue: [" . implode(", ", $eigenvector) . "]\n";
}
?>
